actions/views/channel_sidebar.ts

```diff
@@ -52,3 +55,48 @@ export function createCategory(teamId: string, displayName: string, channelIds?:
         });
     };
 }
+
+// moveChannelInSidebar moves a channel to a given category in the sidebar, but it accounts for when the target index
+// may have changed due to archived channels not being shown in the sidebar.
+export function moveChannelInSidebar(categoryId: string, channelId: string, targetIndex: number) {
+    return (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const newIndex = adjustTargetIndexForMove(getState(), categoryId, channelId, targetIndex);
+
+        return dispatch(moveChannelToCategory(categoryId, channelId, newIndex));
+    };
+}
+
+export function adjustTargetIndexForMove(state: GlobalState, categoryId: string, channelId: string, targetIndex: number) {
+    if (targetIndex === 0) {
+        // The channel is being placed first, so there's nothing above that could affect the index
+        return 0;
+    }
+
+    const category = getCategory(state, categoryId);
+    const filteredChannels = makeGetChannelsForCategory()(state, category);
+
+    if (category.channel_ids.length === filteredChannels.length) {
+        // There are no archived channels in the category, so the targetIndex from react-beautiful-dnd will be correct
+        return targetIndex;
```

Comment:
Do we need to consider channel moving downwards with archive channels?

Reply:
Yeah, all the logic below this is about handling archived channels. If there are no archived channels, then react-beautiful-dnd handles everything itself, so targetIndex will already be correct.